Training/Training/TestReversal.c

```diff
@@ -0,0 +1,131 @@
+      fgets (option, sizeof (option), stdin);
+      char* Result = strchr (option, ch);
+      if (Result == NULL) {
+         while (getchar () != '\n');
+         printf (ANSI_COLOR_RED"Please enter number between 1-5\n\n"ANSI_RESET_ALL);
+      } else {
+         int num = atoi (option);
+         switch (num) {
+            case 1:
+               GetStringInput ();
+               break;
+            case 2:
+               GetIntegerInput ();
+               break;
+            case 3:
+               OutputCheck ();
+               break;
+            case 4:
+               ClrScreen ();
+               break;
+            case 5:
+               return 0;
+               break;
+         }
+      }
+   }
+   return 0;
+}
+
+void GetStringInput () {
+   char ch = '\n';
+   char phrase[MAX_CHAR];
+   printf ("\nEnter a Phrase: ");
+   char* j = fgets (phrase, sizeof (phrase), stdin);
+   char* strResult = strchr (phrase, ch); //To check whether input exceeds limit
+   if (strResult == NULL) while (getchar () != '\n');
+   (j != NULL && phrase[0] != '\n' && strResult != NULL) ? PrintResult (StrIsPalindrome (phrase, 1)) : printf (ANSI_COLOR_RED"INVALID!!!\n\n"ANSI_RESET_ALL);
+}
+
+void GetIntegerInput () {
+   errno = 0;
+   char ch = '\n';
+   char input[MAX_DIGIT];
+   int number = 0;
+   char* endptr;
+   printf ("\nEnter an integer: ");
+   char* i = fgets (input, sizeof (input), stdin);
+   char* intResult = strchr (input, ch); //To check whether input exceeds limit
+   number = strtol (input, &endptr, 10);
+   long long int revNum = 0;
+   long long int* ptr = &revNum;
+   if (intResult == NULL) while (getchar () != '\n');
+   (i == NULL || endptr == input || *endptr != '\n' || input[0] == ' ' || input[0] == '\t') ? printf (ANSI_COLOR_RED"INVALID!!!\n\n"ANSI_RESET_ALL) :
+      (errno == ERANGE) ? printf (ANSI_COLOR_RED"INPUT LIMIT EXCEEDED!!!\n\n"ANSI_RESET_ALL) : PrintResultInt (NumIsPalindrome (number, ptr), revNum);
+}
+
+void PrintResult (int value) {
+   printf (value == 1 ? (ANSI_COLOR_GREEN"It is a Palindrome\n\n"ANSI_RESET_ALL) :
+           value == 0 ? (ANSI_COLOR_RED"It is not a Palindrome\n\n"ANSI_RESET_ALL) : (ANSI_COLOR_RED"INVALID!!!\n\n"ANSI_RESET_ALL));
+}
+
+void PrintResultInt (int value, long long int num) {
+   printf ("Output: %lld\n", num);
+   PrintResult (value);
+}
+
+void OutputCheck () {
+   printf ("\nPALINDROME CHECKER:\n\n--------------INPUT----------------------OUTPUT------------RESULT---");
+   char* strInput[] = { "Don't nod","I did, did I?","Was it a car or a cat I saw?","@@@","RADAR","NASA","CIVIC","NUN","TRUMPF METAMATION" };
+   int strLength = sizeof (strInput) / sizeof (strInput[0]);
+   for (int i = 0; i < strLength; i++) {
+      char* strOutput = "It is not a Palindrome";
+      char* strResult = ANSI_COLOR_RED"FAIL"ANSI_RESET_ALL;
+      int result = StrIsPalindrome (strInput[i], 0);
+      if (result == 1 || result == 0 || result == -1) strResult = ANSI_COLOR_GREEN"PASS"ANSI_RESET_ALL;

```

Comment:
This condition is not needed. There is no point checking for all possible values. Change this wherever possible.